src/machine.rs

```diff
@@ -60,49 +65,55 @@ impl<const N: usize> From<[(Reg, Word); N]> for Registers {
     }
 }
 
-pub trait IO {
-    fn write_string(&mut self, value: &str);
+pub trait Sys: Write {}
+
+pub struct TermSys {
+    out: Stdout,
+}
+
+impl Default for TermSys {
+    fn default() -> Self {
+        Self { out: stdout() }
+    }
+}
+
+impl Write for &mut TermSys {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        self.out.write(buf)
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        Ok(())
```

Comment:
Whoops, this should call `self.out.flush()`; I copied and pasted it from the `TestSys` implementation where flushing (a Vec) is a no-op.